pkg/reconciler/common/utils.go

```diff
@@ -22,24 +22,28 @@ import (
 	"fmt"
 
 	mf "github.com/manifestival/manifestival"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 )
 
-// FetchVersionFromCRD finds the component version from the crd labels, it looks for the
-// label on the first crd it finds after filtering
-// It will return error if crds are not found in manifest or label is not found
-func FetchVersionFromCRD(manifest mf.Manifest, releaseLabel string) (string, error) {
-	crds := manifest.Filter(mf.CRDs)
-	if len(crds.Resources()) == 0 {
-		return "", fmt.Errorf("failed to find crds to get release version")
+// FetchVersionFromConfigMap finds the component version from the ConfigMap data field. It looks
+// for the version key in the ConfigMap and if the ConfigMap or version key is not found
+// then return the error.
+func FetchVersionFromConfigMap(manifest mf.Manifest, configMapName string) (string, error) {
+	configMaps := manifest.Filter(mf.ByKind("ConfigMap"), mf.ByName(configMapName))
+
+	if len(configMaps.Resources()) == 0 {
+		return "", fmt.Errorf("failed to find ConfigMap %s to get release version", configMapName)
 	}
 
-	crd := crds.Resources()[0]
-	version, ok := crd.GetLabels()[releaseLabel]
-	if !ok {
-		return version, fmt.Errorf("failed to find release label on crd")
+	versionConfigMap := configMaps.Resources()[0]
+	dataObj, _, _ := unstructured.NestedStringMap(versionConfigMap.Object, "data")
+	version := dataObj["version"]
+
+	if version != "" {
+		return version, nil
 	}
 
-	return version, nil
+	return "", fmt.Errorf("failed to find version data field on ConfigMap %s", configMapName)
 }
```

Comment:
@vinamra28 when version is unknown (configmap not exist, or version field not exist) let us return `unknown`, instead of returning error. 

Let us not return error and block/break installation for this reason. 🧑‍💻

Reply:
@nikhil-thomas maybe we can return error from here and handle that in TektonPipeline and TektonTriggers controller. This way our error would be logged and we can set version to unknown over there. WDYT?

Reply:
like :arrow_down: 
```go
                var releaseVersion string = "unknown"

		// Read the release version of pipelines
		releaseVersion, err = common.FetchVersionFromConfigMap(manifest, versionConfigMap)
		if err != nil {
			logger.Infow("failed to read release version from ConfigMap", err)
		}
```

Reply:
this could work, 
but i would make it an internal implementation detail of `releaseVersion()`, because this error handling doesn't add more claritiy to the reconciler flow. 🧑‍💻

Reply:
why do we wanna show unknown? in what case that would happen?
nightly, release version, dev release yaml 
all have version set with some value.
then if version is not set then doesn't mean something is wrong?

Reply:
> then if version is not set then doesn't it mean something is wrong?
yes, but as this won't affect any critical functionality. 
can we just do a warning. is the error and break necessary.

Reply:
in addition, we will see the `Unknown` value in the CR status, i think that could be the necessary indicator that something is wrong 😇

Reply:
> this won't affect any critical functionality

if the release yaml is missing something then it can
i don't mind keeping it this way 😄  ... but if we assume version is missing i.e. something is wrong with release yaml and still we allow everything to install then we might break an existing working pipelines, triggers
if we stop then only operator would fail !

Reply:
we can "type" the error and have a `IsUnknow` error check. From this part of the code perspective, it make sense to return an error. The consumer of this is the one with the responsability to handle it.